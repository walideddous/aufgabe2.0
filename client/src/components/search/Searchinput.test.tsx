import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Searchinput from "./SearchInput";

const setUp = (props: any) => {
  const component = mount(<Searchinput {...props} />);
  return component;
};

describe("Searchinput component", () => {
  let mountWrapper: any;
  let props = {
    stations: [],
    handleSelectAutoSearch: jest.fn(),
  };

  beforeEach(() => {
    mountWrapper = setUp(props);
  });
  it("match snapShot with the Seachinput component", () => {
    expect(toJSON(mountWrapper)).toMatchSnapshot();
  });
});
