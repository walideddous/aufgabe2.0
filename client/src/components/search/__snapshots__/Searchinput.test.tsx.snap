// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchinput component Should match snapShot with the Seachinput component 1`] = `
<div
  style={
    Object {
      "position": "absolute",
      "right": "9vh",
      "top": "8vh",
      "width": "75%",
      "zIndex": 400,
    }
  }
>
  <ForwardRef(AutoComplete)
    id="search"
    onChange={[Function]}
    onSelect={[Function]}
    open={false}
    placeholder="Enter stops name"
    style={
      Object {
        "width": "100%",
      }
    }
    value=""
  >
    <Option
      key="5f6203bb0d5658001cd8f85a"
      value="Basel"
    >
      <i
        className="fas fa-subway"
        style={
          Object {
            "color": "#1890ff",
            "margin": "0 10px",
          }
        }
      />
      Basel
    </Option>
    <Option
      key="5f6203bb0d5658001cd8f85b"
      value="Lyon"
    >
      <i
        className="fas fa-subway"
        style={
          Object {
            "color": "#1890ff",
            "margin": "0 10px",
          }
        }
      />
      Lyon
    </Option>
  </ForwardRef(AutoComplete)>
</div>
`;
